pennylane/devices/execution_config.py

```diff
@@ -67,6 +83,9 @@ class ExecutionConfig:
     derivative_order: int = 1
     """The derivative order to compute while evaluating a gradient"""
 
+    mcm_config: MCMConfig = Union[MCMConfig(), dict]
```

Comment:
I'm a little confused by the `= Union[MCMConfig(), dict]`.  Mind adding a comment as to what is going on here?

Reply:
Screwed it up. It should be `mcm_config: Union[MCMConfig, dict] = MCMConfig()`. I changed the type hint from `Optional[MCMConfig]` to `MCMConfig` and I got this pylint warning `Non-mapping value self.mcm_config is used in a mapping context (not-a-mapping)`.